content/blog/2025-05-13-bril-concurrency-extension.md

```diff
@@ -0,0 +1,163 @@
++++
+title = "Sailing Bril’s IR into Concurrent Waters"
+[extra]
+bio = """
+  Ananya Goenka is an undergraduate studying CS at Cornell. While she isn’t nerding out over programming languages or weird ISA quirks, she’s writing for Creme de Cornell or getting way too invested in obscure books.
+"""
+[[extra.authors]]
+name = "Ananya Goenka"
+link = "https://ananyagoenka.github.io"
++++
+
+1\. Goal and Motivation
+--------------------------------------
+
+My project aims to add native shared parallelism concurrency support to [Bril](https://github.com/sampsyo/bril). It adds two simple yet powerful instructions, `spawn` and `join`, which let you launch and synchronize threads directly in Bril programs. Beyond just powering up the IR with threads, I wanted to lay the groundwork for real optimizations by writing a thread-escape analysis pass: a static check that flags heap allocations which truly stay local to a single thread, so future compiler passes can skip unnecessary synchronization or RPC overhead.  
+
+2\. Design Decisions and Specification
+--------------------------------------
+
+### 2.1 Instruction Semantics
+
+We introduced two new opcodes:
+
+*   { "op": "spawn", "dest": "t", "type": "thread", "funcs": \["worker"\], "args": \["x", "y"\] }
+    
+*   **join**: An effect operation that takes a single thread handle argument and blocks until the corresponding thread completes.
+    
+To prevent clients from forging thread IDs, we defined a new primitive type thread in our TypeScript definitions (bril.ts). This opaque type ensures only genuine spawn instructions can produce valid thread handles.
```

Comment:
More code backticks are in order:
* thread -> `thread`
* bril.ts -> `bril.ts`

…and consider doing this to filename and TypeScript symbol names throughout.